dlfcn.c

```diff
@@ -463,6 +465,92 @@ char *dlerror( void )
     return error_buffer;
 }
 
+#ifdef __USE_GNU

```

Comment:
It is really needed to hide dladdr implementation when `__USE_GNU` is not defined? dladdr is available also on non-GNU systems (as you already checked during investigation), so for me it does not look correct to restrict it for GNU-only.